docs/modules/ROOT/pages/references/architecture/sli_reporting.adoc

```diff
@@ -0,0 +1,216 @@
+= SLI & SLO Reporting Service
+
+== Problem Statement
+VSHN has a defined set of xref:oc4:ROOT:explanations/decisions/customer-facing-slo.adoc[customer facing SLOs] which are monitored automatically on VSHN managed OpenShift.
+However, the SLI numbers reported by monitoring don't directly correspond to numbers that can be compared against a contractual SLA.
+
+VSHN isn't contractually accountable for service interruptions that are a direct result of 3rd party service interruptions outside of VSHN's control.
+Even so, a 3rd party service interruption can be picked up by VSHN's SLI monitoring, since it can result in a real service interruption.
+
+For customer-facing SLI and SLO reporting, we need a way to exclude any service interruptions that are caused by a 3rd party.
+This exclusion should be automatic to reduce the risk of human error during calculation.
+
+== High Level Goals
+
+* We've a well-defined source of truth for adjusted SLIs and SLOs.
+* SLOs and remaining error budget can be viewed at any time without doubt about the correctness of the numbers
+* 3rd party outages are detected automatically where possible, but can also be entered manually.
+
+== Non-Goals
+
+* Pretty user interface
+* Automatic reporting of SLO burndown to customers
+
+== Implementation
+
+We write a new service that stores external downtime windows and calculates the adjusted SLIs.
+The service provides a simple REST API to manage 3rd party downtime windows.
+Downtime windows are stored in a simple database.
+
+The service queries our existing monitoring (Mimir) for the SLO monitoring data, and adjusts the results to exclude any service interruptions that fall into a 3rd party downtime window.
+A simple query API can be used to query for the updated SLOs.
+
+The service runs on VSHN's centralized infrastructure, same as VSHN's central monitoring.
+Thanks to this, the service doesn't need to be exposed to the internet.
+
+
+=== Service architecture
+
+The SLO reporting service consists of two architectural components: the downtime window API, and the data querying API.
+While both components are largely independent and could easily be split into separate services, we gain no real benefit from doing so: we require none of the benefits provided by a microservice-style architecture.
```

Comment:
I am actually a bit on the fence whether this makes sense to keep as a monolithic service; the two separate functions are really very independent.

My current reasoning for keeping them together is easier development & deployment. But an argument could be made that if they're separate services, they could be developed in parallel and each part could be swapped out more easily (particularly the querying part might change in the future).

I know how yall like "neat" architecture. Wdyt? ^^